lib/value-pairs/value-pairs.c

```diff
@@ -601,34 +601,42 @@ vp_walker_skip_sdata_enterprise_id(const gchar *name, gint pos)
   do
     {
       /* skip @ or . */
-      pos++;
-      pos += strspn(&name[pos], "0123456789");
+      ++name;
+      name += strspn(name, "0123456789");
     }
-  while (name[pos] == '.' && isdigit(name[pos + 1]));
-  return pos;
+  while (*name == '.' && isdigit(*(name + 1)));
+  return name;
 }
 
 static GPtrArray *
 vp_walker_split_name_to_tokens(vp_walk_state_t *state, const gchar *name)
 {
-  gint pos, token_start = 0, name_len = strlen(name);
+  const gchar *token_start = name;
+  const gchar *token_end = name;
+
   GPtrArray *array = g_ptr_array_sized_new(VP_STACK_INITIAL_SIZE);
 
-  pos = 0;
-  do
+  while (*token_end)
     {
-      token_start = pos;
-      pos += strcspn(&name[pos], "@.");
-      if (name[pos] == '@')
-        pos = vp_walker_skip_sdata_enterprise_id(name, pos);
-
-      if (name[pos] == '.' || pos == name_len)
+      if (*token_end == '@')
```

Comment:
My response to this is that it looks good in both forms to me. It's still
plus one from me.

We're waiting for another viewpoint on this question and/or a plus one
overall, so we can finally merge it.

On Thu, Mar 17, 2016 at 10:56 AM, MÓZES Ádám István <
notifications@github.com> wrote:

> In lib/value-pairs/value-pairs.c
> <https://github.com/balabit/syslog-ng/pull/969#discussion_r56479230>:
>
> >      {
> > -      token_start = pos;
> > -      pos += strcspn(&name[pos], "@.");
> > -      if (name[pos] == '@')
> > -        pos = vp_walker_skip_sdata_enterprise_id(name, pos);
> > -
> > -      if (name[pos] == '.' || pos == name_len)
> > +      if (*token_end == '@')
>
> The body of the loop can be rewritten to a simple switch-case statement.
> Should it be done?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/balabit/syslog-ng/pull/969/files/3043b05422498b5f3345f0bc8e7c54cd796a5009#r56479230>
>